crates/aptos/src/common/types.rs

```diff
@@ -1318,6 +1334,26 @@ impl TransactionOptions {
             gas_unit_price
         };
 
+        // Get sequence number for account
+        let (account, state) = get_account_with_state(&client, sender_address).await?;
+        let sequence_number = account.sequence_number;
+
+        // Retrieve local time, and ensure it's within an expected skew of the blockchain
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map_err(|err| CliError::UnexpectedError(err.to_string()))?
+            .as_secs();
+        let now_usecs = now * US_IN_SECS;
+        if now_usecs > state.timestamp_usecs + ACCEPTED_CLOCK_SKEW_US {
+            return Err(CliError::UnexpectedError(format!("Failed to send transaction, clock is more than {} seconds ahead of the blockchain {}", ACCEPTED_CLOCK_SKEW_US, state.timestamp_usecs / US_IN_SECS )));
```

Comment:
Do we want to fail on clock skew or just take the fullnode's clock? Also, is there potentially problem with the fullnode's clock being skewed from the validators'?

Reply:
I just changed it from the fullnode's clock from a different discussion.

It should use the local clock, this is just to provide some useful feedback for most users.  Similar to most applications that reject things when the clock is off by a significant time period.

I would argue that if your clock is way off, you probably don't want to send transactions.

Now, yes the full node could be behind, so maybe we increase the acceptable skew.